register_maps/econ-to-header.py

```diff
@@ -0,0 +1,125 @@
+import sys
+import yaml
+from pathlib import Path
+
+def count_bits(mask):
+    """Count number of 1s in the binary representation of mask."""
+    return bin(mask).count("1")
+
+def safe_int(val):
+    """Convert YAML value to int whether it's a string like '0x0398' or already an int."""
+    if isinstance(val, str):
+        return int(val, 16) if val.startswith("0x") else int(val)
+    return int(val)
+
+def make_register_locations(address, mask, shift, size_byte):
+    """Split a register into 8-bit chunks if it spans multiple bytes.
+    syntax reminder: RegisterLocation(reg, min_bit, n_bits):
```

Comment:
Wouldn't it be 32-bit (4 byte) chunks because the registers are 32-bits wide?

The "RegisterLocation" developed for the HGCROC focuses on 8-bit wide registers since that's how big the registers are in the HGCROC, but I think the same struct will work with higher offsets/sizes for the larger registers on the ECOND.

Reply:
No, no, let me clarify that the registers are also 8-bit wide (and some parameters span more than 1 byte).